### PR TITLE
Revision of ref manual Chapter 52 (fp semigroups and monoids) fixes #1397

### LINKED ISSUE #1397: ElementOfFpMonoid undocumented
It appears that ElementOfFpMonoid works as expected (the same way as ElementOfFpSemigroup) as a sort of inverse to UnderlyingElement, and I expect to use this operation in the Kan update under preparation.  But the operation is undocumented, and perhaps that is true of many of the functions in fpmon.gd.  Is there good reason for this?  Or is it reasonable to add brief entries in the chapter "Finitely Presented Semigroups and Monoids"?  In my gapdev of May 25th: 
gap> ?ElementOfFpMonoid
Help: no matching entry found
Help: 'ElementOfFpMonoid' is currently undocumented.
      For details, try ?Undocumented Variables

### PR DESCRIPTION
**Pull request by cdwensley**
Section 52.5 of the Reference Manual used to say "The functionality available for finitely presented monoids is essentially the same as that available for finitely presented semigroups, and thus the previous sections apply (with the obvious changes) to finitely presented monoids. 
These changes have been made because the development version of Kan is using ElementOfFpMonoid so it is desirable to have it documented.  The full list of functions newly documented is: FactorFreeMonoidByRelations; IsomorphismFpMonoid; ElementOfFpMonoid; FreeMonoidOfFpMonoid; RelationsOfFpMonoid.

### CODE REVIEW

**Comment by cdwensley** [comment]
It appears to me that it would be difficult to approve this request without a sight of the chapter as a pdf and that this ought to be possible by attaching a file here:
[chap52.pdf](https://github.com/gap-system/gap/files/1095323/chap52.pdf)
The changes start with section 52.2 on page 792.

File: lib/rwssmg.gd [hunk 1]
```diff
@@ -38,7 +38,7 @@
 ##  an attribute storing object (see <Ref Sect="Representation"/>).
 ##  <P/>
 ##  <Example><![CDATA[
-##  gap> f := FreeSemigroup( "a" , "b" );;
+##  gap> f := FreeSemigroup( "a", "b" );;
 ##  gap> a := GeneratorsOfSemigroup(f)[1];;
```
Comment (james-d-mitchell) [inline lib/rwssmg.gd]: `a := f.1;` is shorter

Comment (cdwensley) [inline lib/rwssmg.gd]: Indeed.  Commands of the form "a := GeneratorsOfSemigroup(f)[1];" appear throughout the examples in this chapter.  I will make some more changes.

**Comment by cdwensley** [comment]
Hopefully nothing more to change.  Revised 52.1 because the example given as part of the FpGrpMonSmgOfFpGrpMonSmgElement subsection in fact used only the three pairs of functions which came before that subsection.

File: doc/ref/fpsemi.xml [hunk 2]
```diff
@@ -20,9 +21,9 @@ by a set of relations (a generating set for the congruence), i.e., a set of
 pairs of words in the free semigroup.
 <P/>
 <Example><![CDATA[
-gap> f:=FreeSemigroup("a","b");;
-gap> x:=GeneratorsOfSemigroup(f);;
-gap> s:=f/[ [x[1]*x[2],x[2]*x[1]] ];
+gap> f := FreeSemigroup( "a", "b" );;
+gap> x := GeneratorsOfSemigroup(f);;
+gap> s := f/[ [x[1]*x[2], x[2]*x[1]] ];
```
Comment (james-d-mitchell) [inline doc/ref/fpsemi.xml]: This should either be `f / [ [ x[1] * x[2], x[2] * x[1] ] ];` or `f / [[x[1] * x[2], x[2] * x[1]]];` but not a combination of both.

File: doc/ref/fpsemi.xml [hunk 3]
```diff
@@ -34,12 +35,12 @@ Finitely presented monoids are obtained by factoring a free monoid by
 a set of relations, i.e. a set of pairs of words in the free monoid. 
 <P/>
 <Example><![CDATA[
-gap> f:=FreeMonoid("a","b");;
-gap> x:=GeneratorsOfMonoid(f);
+gap> f := FreeMonoid( "a", "b" );;
+gap> x := GeneratorsOfMonoid(f);
```
Comment (james-d-mitchell) [inline doc/ref/fpsemi.xml]: `x := GeneratorsOfMonoid( f )` if we are including spaces.

Comment (cdwensley) [inline doc/ref/fpsemi.xml]: I'm unfamiliar with the spacing conventions in the reference manual, assuming such exist and are rigorously adhered to, but will investigate.

Comment (james-d-mitchell) [inline doc/ref/fpsemi.xml]: I don't know if they are adhered to or not, but in this example whitespace is introduced inconsistently. This is probably not very important, but it'd be nice if it was done consistently.

Comment (cdwensley) [inline doc/ref/fpsemi.xml]: Agreed.

**Comment by cdwensley** [comment]
To my mind the examples in Chapter 52 are now reasonably consistent, 
but others may find to the contrary? 
Got into a terrible GitHub muddle which Max sorted out for me via Slack.

**Comment by james-d-mitchell** [comment]
Agreed, I think this is an improvement.

**Pull request merged**